apps/oc_chef_authz/src/oc_chef_cookbook_artifact_version.erl

```diff
@@ -0,0 +1,200 @@
+    Metadata = compress(cookbook_artifact_metadata,
+                        ej:get({<<"metadata">>}, CAVData)),
+    SerializedObject = compress(chef_cookbook_artifact_version,
+                        ej:delete({<<"metadata">>}, CAVData)),
+    Checksums = chef_cookbook_version:extract_checksums(CAVData),
+
+    Old#oc_chef_cookbook_artifact_version{identifier = Identifier,
+                                          metadata = Metadata,
+                                          serialized_object = SerializedObject,
+                                          name = Name,
+                                          checksums = Checksums}.
+
+compress(Type, Data) ->
+    chef_db_compression:compress(Type, chef_json:encode(Data)).
+
+-spec new_record(object_id(), object_id(), ejson_term()) -> #oc_chef_cookbook_artifact_version{}.
+new_record(OrgId, AuthzId, CAVData) ->
+    Now = chef_object_base:sql_date(now),
+    Base = #oc_chef_cookbook_artifact_version{authz_id = AuthzId,
+                                              org_id = OrgId,
+                                              created_at = Now},
+    update_from_ejson(Base, CAVData).
+
+set_created(#oc_chef_cookbook_artifact_version{} = Object, ActorId) ->
+    Object#oc_chef_cookbook_artifact_version{created_by = ActorId}.
+
+set_updated(#oc_chef_cookbook_artifact_version{}, _ActorId) ->
+    erlang:error(not_supported).
+
+fields_for_update(#oc_chef_cookbook_artifact_version{}) ->
+    erlang:error(not_supported).
+
+fields_for_fetch(#oc_chef_cookbook_artifact_version{org_id = OrgId,
+                                                    name = Name,
+                                                    identifier = Identifier}) ->
+    [OrgId, Name, Identifier].
+
+%% @doc Fetches a single record
+-spec fetch(DbContext :: chef_db:db_context(),
+            OrgId :: object_id(),
+            Name :: binary(),
+            Identifier :: binary()) ->
+                   #oc_chef_cookbook_artifact_version{} |
+                   not_found |
+                   {error, term()}.
+fetch(DbContext, OrgId, Name, Identifier) ->
+    chef_db:fetch(#oc_chef_cookbook_artifact_version{org_id = OrgId,
+                                                     name = Name,
+                                                     identifier = Identifier},
+                  DbContext).
+
+fetch(#oc_chef_cookbook_artifact_version{} = Record, CallbackFun) ->
+    chef_object:default_fetch(Record, CallbackFun).
+
+list(#oc_chef_cookbook_artifact_version{org_id = OrgId}, CallbackFun) ->
+    CallbackFun({list_query(), [OrgId], rows}).
+
+record_fields() ->
+    record_info(fields, oc_chef_cookbook_artifact_version).
+
+update(#oc_chef_cookbook_artifact_version{}, _CallbackFun) ->
+	erlang:error(not_supported).
+
+flatten(#oc_chef_cookbook_artifact_version{} = Rec) ->
+    %% doesn't have a char(32) ID like other chef objects, but
+    %% rather a DB serial, so we can afford the ID to be undefined
```

Comment:
I had to track this one down to make sure I understood why we were excluding it  -  Perhaps a comment  to that effect of "the ID for this field is generated by the database, so we don't provide it as an input to our create query"?